tests/epyccel/test_arrays.py

```diff
@@ -3808,60 +3818,144 @@ def test_array_view_steps_F_1D_1(language):
 
     f1 = arrays.array_view_steps_F_1D_1
     f2 = epyccel(f1, language = language)
-    assert np.array_equal(f1(a), f2(a))
+    check_array_equal(f1(a), f2(a))
 
 def test_array_view_steps_F_1D_2(language):
     a = arrays.a_1d_f
 
     f1 = arrays.array_view_steps_F_1D_2
     f2 = epyccel(f1, language = language)
-    assert np.array_equal(f1(a), f2(a))
+    check_array_equal(f1(a), f2(a))
 
 ##==============================================================================
 ## TEST ARRAY VIEW STEPS ARRAY INITIALIZATION ORDER F 2D
 ##==============================================================================
 
-@pytest.mark.parametrize( 'language', [
-        pytest.param("c", marks = pytest.mark.c),
-        pytest.param("fortran", marks = [pytest.mark.skip(reason="fortran bug, issue #1257"),
-                                        pytest.mark.fortran])
-    ]
-)
-
 def test_array_view_steps_F_2D_1(language):
     a = arrays.a_2d_f
 
     f1 = arrays.array_view_steps_F_2D_1
     f2 = epyccel(f1, language = language)
-    assert np.array_equal(f1(a), f2(a))
-
-@pytest.mark.parametrize( 'language', [
-        pytest.param("c", marks = pytest.mark.c),
-        pytest.param("fortran", marks = [pytest.mark.skip(reason="fortran bug, issue #1257"),
-                                        pytest.mark.fortran])
-    ]
-)
+    check_array_equal(f1(a), f2(a))
 
 def test_array_view_steps_F_2D_2(language):
     a = arrays.a_2d_f
 
     f1 = arrays.array_view_steps_F_2D_2
     f2 = epyccel(f1, language = language)
-    assert np.array_equal(f1(a), f2(a))
-
-@pytest.mark.parametrize( 'language', [
-        pytest.param("c", marks = pytest.mark.c),
-        pytest.param("fortran", marks = [pytest.mark.skip(reason="fortran bug, issue #1257"),
-                                        pytest.mark.fortran])
-    ]
-)
+    check_array_equal(f1(a), f2(a))
 
 def test_array_view_steps_F_2D_3(language):
     a = arrays.a_2d_f
 
     f1 = arrays.array_view_steps_F_2D_3
     f2 = epyccel(f1, language = language)
-    assert np.array_equal(f1(a), f2(a))
+    check_array_equal(f1(a), f2(a))
+
+#==============================================================================
+# TEST: Array with ndmin argument
+#==============================================================================
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = pytest.mark.fortran),
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("python", marks = [
+            pytest.mark.skip(reason=("Template results in wrong ordered arrays")),
+            pytest.mark.python]
+        )
```

Comment:
Can you please explain what goes wrong in the Python printer? A corresponding open issue should be referenced here.

Reply:
The problem is that the code is different for the different orders. This is because the order is printed explicitly. This could be avoided by saving the initial order as we now do for the dtype. But I'm not convinced it is worth it as this use case is not common.

There is no existing relevant issue. There was one but it was converted to discussion #1334. This could be referenced but I am not sure that it helps clarify the reason for the xfail. Additionally when a fix is chosen and implemented it should fix all Python xfail marks so referencing the discussion shouldn't make any difference for finding xfails to remove

Reply:
Could you please give me an example of how the code differs based on the order of the input array?

Reply:
Eg here:
```python
@template('T', ['int[:]', 'int[:,:]', 'int[:,:,:]', 'int[:,:](order=F)', 'int[:,:,:](order=F)']) 
 def array_ndmin_1(x : 'T'): 
     from numpy import array 
     y = array(x, ndmin=1) 
     return y
```
The argument types `int[:,:]`, and `int[:,:](order=F)` don't work at the same time as the code generated for the first type is something like:
```python
 def array_ndmin_1(x : 'int[:,:]'): 
     from numpy import array 
     y = array(x, dtype=x.dtype, order='C') 
     return y
```
(You may have to double check the actual generated code). NumPy allows properties like the order to be changed when copying data

Reply:
I see. The problem is that we compute the value of `order` explicitly, instead of keeping the original argument `order='K'`, which is indeed the default value for `numpy.array`.

Reply:
Yes. This makes our life a lot easier when working out the order of any results for printing in C/Fortran, but in cases like this it causes differences in printing